### ISSUE TITLE
Data Export failing

### ISSUE THREAD

**Issue by codydirks** [zooniverse/edu-api-front-end#120]
After requesting a data export for either the GZ101 or Hubble's Law activity, trying to obtain the exported data yields a "No export available for this classroom and assignment or the export failed" error. This error appears for both the CSV and Google Drive export options.

This was initially reported by a user trying to test out the system using a second account as a test student. I was able to recreate the error in a test classroom with myself as the only student.

Screenshot showing export was successfully processed and appears to be ready for download:
<img width="1438" alt="screen shot 2018-04-04 at 11 00 58 am" src="https://user-images.githubusercontent.com/20101396/38319511-8f9cc144-37f7-11e8-91de-afa0d700bd1b.png">

Resulting error message after either CSV or Google Drive options are clicked.
<img width="1441" alt="screen shot 2018-04-04 at 11 01 13 am" src="https://user-images.githubusercontent.com/20101396/38319545-a63acd60-37f7-11e8-8acb-046c2b561d46.png">

**Comment by trouille** [comment]
potentially a caesar issue.

**Comment by trouille** [comment]
@codydirks Could you try the system to see if it works with 2 students, 3 students, 4 students, or 5 students? We're trying to understand if this happens only if there's only 1 student. 

From Julia's email, it seems she never tested it with all her students? Granted, I can imagine a lot of instructors would do this kind of test, so it will be important to make sure it works for only 1 student as well.

**Comment by codydirks** [comment]
@trouille Is there a simple way to create fake users for the sake of testing (other than registering a bunch of new Zooniverse accounts with various emails)? I may ask the intro2astro channel to contribute some test classifications, but I'd like more control over exactly how many users are in the classroom.

**Comment by trouille** [comment]
@srallen do you have a good response (re: creating multiple users for testing purposes)?

**Comment by srallen** [comment]
I'm not really sure we need to test group sizes. I had commented to @trouille that if this was an issue just in testing that it's a low priority vs an issue in a classroom setting which would be high priority. I didn't mean to imply that group sizes are a variable in causing the issue. This failure sounds like a caesar issue. Basically the export to Google Sheets or download will fail because it does some parsing of the original caesar export CSV to make it more human-readable friendly. It'll fail if the original caesar CSV is empty. I think we've seen empty caesar CSVs when there are classifications made on the project outside of the classroom group. What would be helpful for testing:

A new classroom with a student joined. Have that student user classify not in the group. Then have that student classify in the classroom group. Request the export.

@amyrebecca is there any way we can check on the reductions for workflow ids 5521 and 5522 to see if there are any recent issues? Or is that difficult without a user group id?

**Comment by srallen** [comment]
@amyrebecca I'm not really sure how urgent this is since it sounds like it was a bug reported in a testing scenario initially. I can check with @trouille

**Comment by trouille** [comment]
Correct, it's not urgent (i.e., it can wait for a couple weeks). Tommy has provided the instructors 'sample' files to use with their classes. We do want to fix this in May so that any summer courses that want to use these materials can do that, starting in June. Just knowing what might be causing the issue is very helpful. I'll touch base in May with @amyrebecca to find time at that point to clear out the old extracts and do some tests with @codydirks at that time. Thanks all.

**Comment by amyrebecca** [comment]
Once this pull request is merged, the behavior should be a lot better: https://github.com/zooniverse/caesar/pull/286

**Comment by srallen** [comment]
@codydirks I just tested this to confirm that it is fixed. I created a new classroom and had two separate tester accounts join as students. I had one student logged out classify on the galaxy classification project and the other classify logged in. The logged out student's classifications do not count toward the counts nor are included in the aggregated exports, but it is not breaking the exports any longer. 

Could you confirm that this appears to be fixed for new classrooms? For old classrooms, we may need to clear out some bad data out of caesar. Do you know if there are any classrooms that have failing exports that we need to clean the data for?

**Comment by srallen** [comment]
Closing and will reopen if there is still an issue.
